### PR TITLE
Add Rust analyzer support

### LINKED ISSUE #520: Add Rust Language Server
https://rust-analyzer.github.io/

### PR DESCRIPTION
**Pull request by sonnyp**
Fixes #520 

Only supports syntax diagnostics for now. I made a follow up for types https://github.com/workbenchdev/Workbench/issues/881

## How to test

* Open Workbench
* From the Library, open a demo in Rust
* Make a syntax error in Rust code
* Wait a minute or so (rust-analyzer takes a while to initialize)

Diagnostics will appear for syntax errors. 

image

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -94,6 +94,12 @@ export const languages = [
     document: null,
     default_file: "code.rs",
     index: 2,
+    // TODO: remove verbose
+    language_server: ["rust-analyzer", "--verbose"],
+    formatting_options: {
+      ...formatting_options,
+      tabSize: 2,
```
Comment (Hofer-Julian) [inline src/common.js:101]: We currently use rustfmt's defaults here, which specificies a tab size of 4.
Unless there's a good reason, we should stick with that.

File: src/langs/rust/RustDocument.js [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
 
-  const [success, stdout, stderr] = rustfmtProcess.communicate_utf8(text, null);
+  async format() {
+    // https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_formatting
+    const text_edits = await this.lspc.request("textDocument/formatting", {
+      textDocument: {
+        uri: this.file.get_uri(),
+      },
+      options: {
+        tabSize: 2,
```
Comment (Hofer-Julian) [inline src/langs/rust/RustDocument.js:20]: Same here

**Comment by Hofer-Julian** [comment]
Considering that there's soon a release, I wonder if we should merge this PR (after a small clean up).

Showing syntax errors is already quite useful, and we could look into diagnostics in a follow-up PR.

**Review by lw64 — APPROVED** [review]

**Review by Hofer-Julian — APPROVED** [review]
Works as expected 🦀
Great work @sonnyp!

**Pull request merged**